File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        maxind,maxi=-1,0
        for i in range(len(nums)):
            if(nums[i]>maxi):
                maxi= nums[i]
                maxind=i
        for num in nums:
            if(maxi<2*num and num!=maxi):
                return -1
        return maxind

File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int maxind=-1,max=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]>max){
                max=nums[i];
                maxind=i;
            }
        }
        for(int num:nums){
            if(max<2*num&&num!=max ){
                return -1;
            }
        }
        return maxind;
    }
}